src/olympia/api/authentication.py

```diff
@@ -20,16 +21,23 @@ class JSONWebTokenAuthentication(UpstreamJSONWebTokenAuthentication):
     """
     DRF authentication class for JWT header auth.
     """
-    def authenticate_credentials(self, request):
+
+    def authenticate_credentials(self, payload):
         """
         Mimic what our ACLMiddleware does after a successful authentication,
         because otherwise that behaviour would be missing in the API since API
         auth happens after the middleware process request phase.
         """
-        result = super(
-            JSONWebTokenAuthentication, self).authenticate_credentials(request)
-        amo.set_user(result)
-        return result
+        try:
+            user = UserProfile.objects.get(pk=payload['user_id'])
+        except UserProfile.DoesNotExist:
```

Comment:
Since this case would be a little bit odd (i.e. not something a monkey could easily trigger), it would be helpful to log which user ID is missing from the system